api/order_products.js

```diff
@@ -0,0 +1,67 @@
+const express = require("express");
+const router = express.Router();
+
+const {
+  destroyOrderProduct,
+  updateOrderProduct,
+  getOrderProductById,
+  getOrderById,
+} = require("../db");
+
+router.patch("/:orderProductId", async (req, res, next) => {
+  const { orderProductId } = req.params;
+  const { quantity, price } = req.body;
+  try {
+    const orderProductById = await getOrderProductById(orderProductId);
+    const order = await getOrderById(orderProductById.orderId);
+    if (!order.creatorId) {
+      const updatedGuestOrderProduct = await updateOrderProduct({
+        id: orderProductId,
+        quantity,
+        price,
+      });
+      res.send(updatedGuestOrderProduct);
+    } else if (order.creatorId === req.user.id) {
+      const updatedUserOrderProduct = await updateOrderProduct({
+        id: orderProductId,
+        quantity,
+        price,
+      });
+      res.send(updatedUserOrderProduct);
+    } else {
+      next({
+        name: "ErrorUpdatingOrderProduct",
+        message: "An error occured when trying to update this order.",
+      });
+    }
+  } catch ({ name, message }) {
+    next({ name, message });
+  }
+});
```

Comment:
i think we can leave out price from our patching, what do u think?

I'm thinking something along the lines of this => the price is initially stored in this table when the product is added to an order, storing the current price of the item in order_products at the time that it's added. this way the price will stay the same even if the price of the product that was added changes afterwards. 

So we'll just use the patch route to update the quantity only probably?

Reply:
Makes sense... and we can always add it back in if we discover we need it, but I agree with your logic. It jives with previous discussions we've had about price locking. Change made and committed.